src/compiler/types.ts

```diff
@@ -1329,6 +1330,7 @@ module ts {
 
         // Values for enum members have been computed, and any errors have been reported for them.
         EnumValuesComputed  = 0x00000080,
+        BlockScopedBindingInLoop = 0x00000100,

```

Comment:
According to ES6 spec, `let` in the loop declaration is also scoped _per iteration_ (so subject to the same downlevel codegen issues). Beware that browsers currently get it wrong (tested in IE11, FF33).